lib/request.js

```diff
@@ -57,8 +57,15 @@ var req = exports = module.exports = {
 
 req.get =
 req.header = function header(name) {
-  var lc = name.toLowerCase();
-
+  var lc = null;
+  
+  // If name wasn't a string, it will crash
+  if(typeof name !== 'string') { 
+    return new TypeError('Parameter "name" needs to be a String but got a ' + (typeof name));
```

Comment:
The error should be `throw`n, not returned. If you are going to go along the pattern of "but go a..." I would suggest using the type detection routing from the rest of the case base, which better handles all the various things that simply return `object`, like for example, `null`.